### PR TITLE
Ncgames 6

### PR DESCRIPTION
**Pull request by Rim0Rad**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -175,11 +163,76 @@ describe('GET /api/review/:review_id/comments', () => {
...
-    })
+/* Stop database and test qurry responses for endpoints*/
+describe("500 Internal Server Error", () => {
+  it("returns 500 for internal server error", () => {
+    if (db) db.end();
+    const getCategories = request(app).get("/api/categories").expect(500);
+    const getReviews = request(app).get("/api/reviews").expect(500);
+    const getReviewsById = request(app).get("/api/reviews/2").expect(500);
+    return Promise.all([getCategories, getReviews, getReviewsById]);
+  });
```
Comment (EddTheDuck) [inline __tests__/app.test.js]: Interesting way of testing for a 500. However we don't really expect you to test for it, You don't have to remove it but feel free to not test for this in future.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -175,11 +163,76 @@ describe('GET /api/review/:review_id/comments', () => {
...
+  it("has property comment_count which is the number of comments with this review_id", () => {
+    const commentCount = [null, 0, 3, 3, 0, 0 ,0 ,0 ,0 ,0 ,0 ,0 ,0, 0 ]
+    return request(app).get("/api/reviews")
+      .then((result) => result.body.reviews)
+      .then((reviews) => {
+        reviews.forEach((review) => {
+            expect(review).toHaveProperty("comment_count");
+            expect(review.comment_count).toBe(commentCount[review.review_id]);
+        });
```
Comment (EddTheDuck) [inline __tests__/app.test.js]: Interesting way of testing for comment count. Very thorough.

**Review by EddTheDuck — APPROVED** [review]
Happy for you to merge

File: __tests__/app.test.js (hunk 1 above)
Comment (Rim0Rad) [reply __tests__/app.test.js]: Is was told to test for 500 by one of the staff

**Pull request merged**